test/test_tensordict.py

```diff
@@ -2488,6 +2488,21 @@ def test_setitem_nested():
     assert (tensordict["a", "b", "c"] == 1).all()
 
 
+def test_setdefault_nested():
```

Comment:
looks good but do you think we can test that for the other tensordict classes too?

Reply:
Good idea, have moved the test up into the main test class and added nested items to test this out on the various different types.